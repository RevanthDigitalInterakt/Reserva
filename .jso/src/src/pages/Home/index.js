  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function get() {
      return _Home.default;
    }
  });
  var _Home = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[1]));
