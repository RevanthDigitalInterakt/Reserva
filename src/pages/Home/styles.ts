import { StyleSheet } from 'react-native';
import configDeviceSizes from '../../utils/configDeviceSizes';
import { COLORS } from '../../base/styles';
import { platformType } from '../../utils/platformType';

const styles = StyleSheet.create({
  topBarDefault: {
    zIndex: 999,
    position: 'absolute',
  },
  transparentTopBar: {
    zIndex: 999,
  },
  whiteTopBar: {
    zIndex: 999,
    position: 'absolute',
  },
  rouletWrapper: {
    width: '100%',
    height: '100%',
    backgroundColor: 'transparent',
    position: 'absolute',
    zIndex: 999,
  },
  webView: {
    width: '100%',
    height: configDeviceSizes.DEVICE_HEIGHT * 0.54,
    backgroundColor: 'rgba(0, 0, 0, 0.45)',
    zIndex: 999,
  },
  loaderWrapper: {
    position: 'absolute',
    width: 50,
    height: 50,
    left: configDeviceSizes.DEVICE_WIDTH / 2 - 25,
    top: (configDeviceSizes.DEVICE_HEIGHT * 0.9) / 2 - 25,
    alignItems: 'center',
    justifyContent: 'center',

  },
  closeButton: {
    position: 'absolute',
    left: '10%',
    top: platformType.IOS ? '10%' : 10,
    paddingHorizontal: 10,
    paddingVertical: 5,
    backgroundColor: COLORS.RED,
  },
  closeButtonText: {
    fontSize: 12,
    color: COLORS.WHITE,
  },
});

export default styles;
