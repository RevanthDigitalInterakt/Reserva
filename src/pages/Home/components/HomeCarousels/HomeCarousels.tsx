import React, { useCallback } from 'react';
import { ActivityIndicator, View } from 'react-native';

import type { HomeCarouselOutput } from '../../../../base/graphql/generated';
import { HomePageSectionTypeEnum } from '../../../../base/graphql/generated';
import { COLORS } from '../../../../base/styles/colors';
import { Box } from '../../../../components/Box/Box';
import testProps from '../../../../utils/testProps';
import { useHomeStore } from '../../../../zustand/useHomeStore';
import HomeBrandsCarousel from '../HomeBrandsCarousel';
import HomeCardsCarousel from '../HomeCardsCarousel';
import HomeMainCarousel from '../HomeMainCarousel';

function HomeCarousels() {
  const { carousels, loading } = useHomeStore(['carousels', 'loading']);

  const renderCarousel = useCallback((item: HomeCarouselOutput) => {
    if (!item.items.length) return null;

    const relationalObject: { [key in HomePageSectionTypeEnum | 'DEFAULT']: () => JSX.Element | null } = {
      [HomePageSectionTypeEnum.Main]: () => <HomeMainCarousel data={item} />,
      [HomePageSectionTypeEnum.Brands]: () => <HomeBrandsCarousel data={item} />,
      [HomePageSectionTypeEnum.Cards]: () => <HomeCardsCarousel data={item} />,
      DEFAULT: () => null,
    };

    return (relationalObject[item.type] || relationalObject.DEFAULT)();
  }, []);

  if (loading) {
    return (
      <Box bg="white" marginY="nano" justifyContent="center">
        <View style={{ height: 100 }} />

        <ActivityIndicator size="small" color={COLORS.BLACK} />
      </Box>
    );
  }

  return (
    <View>
      {carousels.map((carousel) => (
        <View
          {...testProps('carousels_cards')}
          key={`item-${carousel.type}-${carousel.showtime}-${carousel.items.length}`}
        >
          {renderCarousel(carousel)}
        </View>
      ))}
    </View>
  );
}

export default HomeCarousels;
