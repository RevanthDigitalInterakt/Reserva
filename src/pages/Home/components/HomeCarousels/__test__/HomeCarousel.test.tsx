import { MockedProvider } from '@apollo/client/testing';
import {
  render,
  screen,
} from '@testing-library/react-native';
import React from 'react';
import { ThemeProvider } from 'styled-components/native';

import { theme } from '../../../../../base/usereservappLegacy/theme';
import { mockHomeCarouselQuery } from '../../../__test__/Home.mock';
import HomeCarousels from '../HomeCarousels';

// MOCKS
const mockNavigate = jest.fn();

jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({ navigate: mockNavigate }),
}));

jest.mock('../../../../../zustand/useHomeStore', () => ({
  useHomeStore: () => ({
    onLoad: () => {},
    carousels: mockHomeCarouselQuery.result.data.homeCarousels,
  }),
}));

const TestingComponent = (
  <ThemeProvider theme={theme}>
    <MockedProvider
      mocks={[mockHomeCarouselQuery]}
      addTypename={false}
    >
      <HomeCarousels />
    </MockedProvider>
  </ThemeProvider>
);

describe('HomeCarousel', () => {
  beforeEach(async () => {
    jest.useFakeTimers({ legacyFakeTimers: true });
    render(TestingComponent);
  });

  it('should render properly', () => {
    const card = screen.getByTestId('com.usereserva:id/default_carrousel_container');
    expect(card).toBeOnTheScreen();
  });

  it('should match to snapshot', () => {
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
