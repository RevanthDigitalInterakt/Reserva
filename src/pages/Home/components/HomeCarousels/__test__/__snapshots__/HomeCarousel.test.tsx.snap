// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeCarousel should match to snapshot 1`] = `
<View>
  <View
    accessibilityLabel="com.usereserva:id/carousels_cards"
    testID="com.usereserva:id/carousels_cards"
  >
    <View
      accessibilityLabel="com.usereserva:id/default_carrousel_container"
      style={
        Object {
          "flex": 1,
        }
      }
      testID="com.usereserva:id/default_carrousel_container"
    >
      <View
        collapsable={false}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "height": 972,
              "width": 750,
            },
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0)",
              "position": "relative",
            },
            Object {
              "flexDirection": "row",
            },
          ]
        }
        testID="com.usereserva:id/default_carrousel_content"
      >
        <View
          style={
            Array [
              Object {
                "height": 972,
                "position": "absolute",
                "width": 750,
              },
              Object {
                "transform": Array [
                  Object {
                    "translateX": undefined,
                  },
                  Object {
                    "scale": undefined,
                  },
                ],
                "zIndex": undefined,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "height": 972,
                "position": "absolute",
                "width": 750,
              },
              Object {
                "transform": Array [
                  Object {
                    "translateX": undefined,
                  },
                  Object {
                    "scale": undefined,
                  },
                ],
                "zIndex": undefined,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "height": 972,
                "position": "absolute",
                "width": 750,
              },
              Object {
                "transform": Array [
                  Object {
                    "translateX": undefined,
                  },
                  Object {
                    "scale": undefined,
                  },
                ],
                "zIndex": undefined,
              },
            ]
          }
        />
      </View>
      <View
        accessibilityLabel="com.usereserva:id/carrousel_scroll_indicator_container"
        bottom={10}
        flexDirection="row"
        position="absolute"
        style={
          Array [
            Object {
              "bottom": 10,
              "flexDirection": "row",
              "position": "absolute",
              "zIndex": 3,
            },
          ]
        }
        testID="com.usereserva:id/carrousel_scroll_indicator_container"
        zIndex={3}
      >
        <View
          backgroundColor="neutroFrio2"
          flex={1}
          height={2}
          marginLeft="quarck"
          marginRight="quarck"
          onLayout={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "#8A8C8E",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "height": 2,
                "marginLeft": 4,
                "marginRight": 4,
              },
              Object {
                "overflow": "hidden",
              },
            ]
          }
        >
          <View
            accessibilityLabel="com.usereserva:id/carrousel_scroll_indicator_animated_view"
            style={
              Object {
                "backgroundColor": "#D8D9DA",
                "height": "100%",
                "left": 0,
                "position": "absolute",
                "top": 0,
                "transform": Array [
                  Object {
                    "translateX": 5,
                  },
                ],
                "width": "100%",
              }
            }
            testID="com.usereserva:id/carrousel_scroll_indicator_animated_view"
          />
        </View>
        <View
          backgroundColor="neutroFrio2"
          flex={1}
          height={2}
          marginLeft={null}
          marginRight="quarck"
          onLayout={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "#8A8C8E",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "height": 2,
                "marginRight": 4,
              },
              Object {
                "overflow": "hidden",
              },
            ]
          }
        />
        <View
          backgroundColor="neutroFrio2"
          flex={1}
          height={2}
          marginLeft={null}
          marginRight="quarck"
          onLayout={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "#8A8C8E",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "height": 2,
                "marginRight": 4,
              },
              Object {
                "overflow": "hidden",
              },
            ]
          }
        />
      </View>
    </View>
  </View>
  <View
    accessibilityLabel="com.usereserva:id/carousels_cards"
    testID="com.usereserva:id/carousels_cards"
  >
    <RCTScrollView
      accessibilityLabel="com.usereserva:id/brands_flatList"
      contentContainerStyle={
        Object {
          "paddingHorizontal": 10,
          "paddingVertical": 2,
        }
      }
      data={
        Array [
          Object {
            "__typename": "HomeCarouselItemOutput",
            "facets": Array [
              Object {
                "__typename": "ProductFacetOutput",
                "key": "productClusterIds",
                "value": "2875",
              },
            ],
            "image": Object {
              "__typename": "HomeCarouselItemImageOutput",
              "height": null,
              "title": "",
              "url": "https://images.ctfassets.net/6jsfqc13oxv4/2KV8eKVsr4frv5Wcd0oYPr/bbb405ea436cf621a9ed5993a3a9d726/AF_RESERVA_LOGO_PREFERENCIAL_VERSAO_COR_POSITIVA.png",
              "width": null,
            },
            "linkMktIn": null,
            "mkt": false,
            "orderBy": "RELEVANCIA",
            "reference": "collection:2875",
            "reservaMini": false,
          },
          Object {
            "__typename": "HomeCarouselItemOutput",
            "facets": Array [
              Object {
                "__typename": "ProductFacetOutput",
                "key": "productClusterIds",
                "value": "930",
              },
            ],
            "image": Object {
              "__typename": "HomeCarouselItemImageOutput",
              "height": null,
              "title": "",
              "url": "https://images.ctfassets.net/6jsfqc13oxv4/1QoMJOvYTiEXSPp5zSDtUi/cff6ae1381f71a26cce9bfe428dd9638/rsvGO-logos-02.png",
              "width": null,
            },
            "linkMktIn": null,
            "mkt": false,
            "orderBy": "RELEVANCIA",
            "reference": "collection:930",
            "reservaMini": false,
          },
          Object {
            "__typename": "HomeCarouselItemOutput",
            "facets": Array [
              Object {
                "__typename": "ProductFacetOutput",
                "key": "productClusterIds",
                "value": "2605",
              },
            ],
            "image": Object {
              "__typename": "HomeCarouselItemImageOutput",
              "height": null,
              "title": "",
              "url": "https://images.ctfassets.net/6jsfqc13oxv4/6sXahjVEmwE9iUsW8xGS0w/e90491227f8afab77a3d88da8f5e5ac9/AF_RESERVA_MINI_LOGO_SIMPLIFICADO_VERSAO_COR_VERMELHO_POSITIVA_RGB.png",
              "width": null,
            },
            "linkMktIn": null,
            "mkt": false,
            "orderBy": "RELEVANCIA",
            "reference": "collection:2605",
            "reservaMini": false,
          },
          Object {
            "__typename": "HomeCarouselItemOutput",
            "facets": Array [
              Object {
                "__typename": "ProductFacetOutput",
                "key": "productClusterIds",
                "value": "1228",
              },
            ],
            "image": Object {
              "__typename": "HomeCarouselItemImageOutput",
              "height": null,
              "title": "",
              "url": "https://images.ctfassets.net/6jsfqc13oxv4/zt55b6oFguzEyCVuxRTp1/1bc54d91b42556a00babbc0a7ce28040/Reversa.png",
              "width": null,
            },
            "linkMktIn": null,
            "mkt": false,
            "orderBy": "RELEVANCIA",
            "reference": "collection:1228",
            "reservaMini": false,
          },
          Object {
            "__typename": "HomeCarouselItemOutput",
            "facets": Array [
              Object {
                "__typename": "ProductFacetOutput",
                "key": "productClusterIds",
                "value": "1587",
              },
            ],
            "image": Object {
              "__typename": "HomeCarouselItemImageOutput",
              "height": null,
              "title": "",
              "url": "https://images.ctfassets.net/6jsfqc13oxv4/1YN28l3BTXvea0PzgDkiP/55d8e1fa4538beabb474aee2f786c4db/oculos_reserva.png",
              "width": null,
            },
            "linkMktIn": null,
            "mkt": false,
            "orderBy": "RELEVANCIA",
            "reference": "collection:1587",
            "reservaMini": false,
          },
        ]
      }
      decelerationRate="fast"
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEnabled={true}
      scrollEventThrottle={16}
      showsHorizontalScrollIndicator={false}
      snapToAlignment="start"
      stickyHeaderIndices={Array []}
      style={
        Object {
          "marginVertical": 13,
        }
      }
      testID="com.usereserva:id/brands_flatList"
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessibilityLabel="com.usereserva:id/brands_brand_container-0"
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderRadius": 10,
                "display": "flex",
                "elevation": 5,
                "height": 48,
                "justifyContent": "center",
                "marginBottom": 10,
                "marginRight": 10,
                "marginTop": 10,
                "maxHeight": 48,
                "maxWidth": 78,
                "opacity": 1,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "shadowOpacity": 0.15,
                "shadowRadius": 4,
                "width": 78,
              }
            }
            testID="com.usereserva:id/brands_brand_container-0"
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "uri": "https://images.ctfassets.net/6jsfqc13oxv4/2KV8eKVsr4frv5Wcd0oYPr/bbb405ea436cf621a9ed5993a3a9d726/AF_RESERVA_LOGO_PREFERENCIAL_VERSAO_COR_POSITIVA.png",
                }
              }
              style={
                Array [
                  Object {
                    "height": 0,
                    "width": 750,
                  },
                  Object {
                    "height": 20,
                    "maxHeight": 20,
                    "maxWidth": 60,
                    "width": 60,
                  },
                ]
              }
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessibilityLabel="com.usereserva:id/brands_brand_container-1"
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderRadius": 10,
                "display": "flex",
                "elevation": 5,
                "height": 48,
                "justifyContent": "center",
                "marginBottom": 10,
                "marginRight": 10,
                "marginTop": 10,
                "maxHeight": 48,
                "maxWidth": 78,
                "opacity": 1,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "shadowOpacity": 0.15,
                "shadowRadius": 4,
                "width": 78,
              }
            }
            testID="com.usereserva:id/brands_brand_container-1"
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "uri": "https://images.ctfassets.net/6jsfqc13oxv4/1QoMJOvYTiEXSPp5zSDtUi/cff6ae1381f71a26cce9bfe428dd9638/rsvGO-logos-02.png",
                }
              }
              style={
                Array [
                  Object {
                    "height": 0,
                    "width": 750,
                  },
                  Object {
                    "height": 20,
                    "maxHeight": 20,
                    "maxWidth": 60,
                    "width": 60,
                  },
                ]
              }
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessibilityLabel="com.usereserva:id/brands_brand_container-2"
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderRadius": 10,
                "display": "flex",
                "elevation": 5,
                "height": 48,
                "justifyContent": "center",
                "marginBottom": 10,
                "marginRight": 10,
                "marginTop": 10,
                "maxHeight": 48,
                "maxWidth": 78,
                "opacity": 1,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "shadowOpacity": 0.15,
                "shadowRadius": 4,
                "width": 78,
              }
            }
            testID="com.usereserva:id/brands_brand_container-2"
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "uri": "https://images.ctfassets.net/6jsfqc13oxv4/6sXahjVEmwE9iUsW8xGS0w/e90491227f8afab77a3d88da8f5e5ac9/AF_RESERVA_MINI_LOGO_SIMPLIFICADO_VERSAO_COR_VERMELHO_POSITIVA_RGB.png",
                }
              }
              style={
                Array [
                  Object {
                    "height": 0,
                    "width": 750,
                  },
                  Object {
                    "height": 20,
                    "maxHeight": 20,
                    "maxWidth": 60,
                    "width": 60,
                  },
                ]
              }
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessibilityLabel="com.usereserva:id/brands_brand_container-3"
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderRadius": 10,
                "display": "flex",
                "elevation": 5,
                "height": 48,
                "justifyContent": "center",
                "marginBottom": 10,
                "marginRight": 10,
                "marginTop": 10,
                "maxHeight": 48,
                "maxWidth": 78,
                "opacity": 1,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "shadowOpacity": 0.15,
                "shadowRadius": 4,
                "width": 78,
              }
            }
            testID="com.usereserva:id/brands_brand_container-3"
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "uri": "https://images.ctfassets.net/6jsfqc13oxv4/zt55b6oFguzEyCVuxRTp1/1bc54d91b42556a00babbc0a7ce28040/Reversa.png",
                }
              }
              style={
                Array [
                  Object {
                    "height": 0,
                    "width": 750,
                  },
                  Object {
                    "height": 20,
                    "maxHeight": 20,
                    "maxWidth": 60,
                    "width": 60,
                  },
                ]
              }
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessibilityLabel="com.usereserva:id/brands_brand_container-4"
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderRadius": 10,
                "display": "flex",
                "elevation": 5,
                "height": 48,
                "justifyContent": "center",
                "marginBottom": 10,
                "marginRight": 10,
                "marginTop": 10,
                "maxHeight": 48,
                "maxWidth": 78,
                "opacity": 1,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 2,
                },
                "shadowOpacity": 0.15,
                "shadowRadius": 4,
                "width": 78,
              }
            }
            testID="com.usereserva:id/brands_brand_container-4"
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "uri": "https://images.ctfassets.net/6jsfqc13oxv4/1YN28l3BTXvea0PzgDkiP/55d8e1fa4538beabb474aee2f786c4db/oculos_reserva.png",
                }
              }
              style={
                Array [
                  Object {
                    "height": 0,
                    "width": 750,
                  },
                  Object {
                    "height": 20,
                    "maxHeight": 20,
                    "maxWidth": 60,
                    "width": 60,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    accessibilityLabel="com.usereserva:id/carousels_cards"
    testID="com.usereserva:id/carousels_cards"
  >
    <View
      accessibilityLabel="com.usereserva:id/cards_carrousel_container"
      style={
        Array [
          Object {},
        ]
      }
      testID="com.usereserva:id/cards_carrousel_container"
    >
      <RCTScrollView
        accessibilityLabel="com.usereserva:id/cards_carrousel_animated_flat_list"
        bounces={false}
        contentContainerStyle={
          Object {
            "paddingLeft": 4,
            "paddingRight": 4,
          }
        }
        data={
          Array [
            Object {
              "__typename": "HomeCarouselItemOutput",
              "facets": Array [
                Object {
                  "__typename": "ProductFacetOutput",
                  "key": "productClusterIds",
                  "value": "2231",
                },
              ],
              "image": Object {
                "__typename": "HomeCarouselItemImageOutput",
                "height": 898,
                "title": "[Card] Liqui Reserva Camisas 2023",
                "url": "https://images.ctfassets.net/6jsfqc13oxv4/4ezAqZI6EBjBKlHbjC1nRR/e5a1c423c52ad2bd705f06b5336be55f/banner-card_app-camisas.jpg",
                "width": 1114,
              },
              "linkMktIn": null,
              "mkt": false,
              "orderBy": "RELEVANCIA",
              "reference": "collection:2231",
              "reservaMini": false,
            },
            Object {
              "__typename": "HomeCarouselItemOutput",
              "facets": Array [
                Object {
                  "__typename": "ProductFacetOutput",
                  "key": "productClusterIds",
                  "value": "2234",
                },
              ],
              "image": Object {
                "__typename": "HomeCarouselItemImageOutput",
                "height": 898,
                "title": "[Card] Liqui Reserva Calças 2023",
                "url": "https://images.ctfassets.net/6jsfqc13oxv4/2cg232VjWWBOuB0mKL8lB6/999a65cb9f5c55090e44a9c35c20d0b6/banner-card_app-calcas.jpg",
                "width": 1114,
              },
              "linkMktIn": null,
              "mkt": false,
              "orderBy": "RELEVANCIA",
              "reference": "collection:2234",
              "reservaMini": false,
            },
            Object {
              "__typename": "HomeCarouselItemOutput",
              "facets": Array [
                Object {
                  "__typename": "ProductFacetOutput",
                  "key": "productClusterIds",
                  "value": "2232",
                },
              ],
              "image": Object {
                "__typename": "HomeCarouselItemImageOutput",
                "height": 898,
                "title": "[Card] Liqui Reserva Polos 2023",
                "url": "https://images.ctfassets.net/6jsfqc13oxv4/28yQznd8Yw0jugpXqSteow/65a161d411d1920a303a5d5c7a65e424/banner-card_app-polos.jpg",
                "width": 1114,
              },
              "linkMktIn": null,
              "mkt": false,
              "orderBy": "RELEVANCIA",
              "reference": "collection:2232",
              "reservaMini": false,
            },
          ]
        }
        decelerationRate="fast"
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={true}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={16}
        showsHorizontalScrollIndicator={false}
        snapToAlignment="start"
        snapToOffsets={
          Array [
            -48,
            589.5,
            1227,
          ]
        }
        stickyHeaderIndices={Array []}
        style={Object {}}
        testID="com.usereserva:id/cards_carrousel_animated_flat_list"
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              accessibilityLabel="com.usereserva:id/card_container_collection:2231"
              style={
                Array [
                  Object {},
                ]
              }
              testID="com.usereserva:id/card_container_collection:2231"
            >
              <View
                alignItems="center"
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                    },
                  ]
                }
              >
                <View
                  accessibilityLabel="com.usereserva:id/card_button_collection:2231"
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "opacity": 1,
                    }
                  }
                  testID="com.usereserva:id/card_button_collection:2231"
                >
                  <Image
                    height={280}
                    resizeMode="contain"
                    source={
                      Object {
                        "uri": "https://images.ctfassets.net/6jsfqc13oxv4/4ezAqZI6EBjBKlHbjC1nRR/e5a1c423c52ad2bd705f06b5336be55f/banner-card_app-camisas.jpg",
                      }
                    }
                    style={
                      Array [
                        Object {
                          "height": 0,
                          "width": 621.5,
                        },
                        undefined,
                      ]
                    }
                  />
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              accessibilityLabel="com.usereserva:id/card_container_collection:2234"
              style={
                Array [
                  Object {},
                ]
              }
              testID="com.usereserva:id/card_container_collection:2234"
            >
              <View
                alignItems="center"
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                    },
                  ]
                }
              >
                <View
                  accessibilityLabel="com.usereserva:id/card_button_collection:2234"
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "opacity": 1,
                    }
                  }
                  testID="com.usereserva:id/card_button_collection:2234"
                >
                  <Image
                    height={280}
                    resizeMode="contain"
                    source={
                      Object {
                        "uri": "https://images.ctfassets.net/6jsfqc13oxv4/2cg232VjWWBOuB0mKL8lB6/999a65cb9f5c55090e44a9c35c20d0b6/banner-card_app-calcas.jpg",
                      }
                    }
                    style={
                      Array [
                        Object {
                          "height": 0,
                          "width": 621.5,
                        },
                        undefined,
                      ]
                    }
                  />
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              accessibilityLabel="com.usereserva:id/card_container_collection:2232"
              style={
                Array [
                  Object {},
                ]
              }
              testID="com.usereserva:id/card_container_collection:2232"
            >
              <View
                alignItems="center"
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                    },
                  ]
                }
              >
                <View
                  accessibilityLabel="com.usereserva:id/card_button_collection:2232"
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "opacity": 1,
                    }
                  }
                  testID="com.usereserva:id/card_button_collection:2232"
                >
                  <Image
                    height={280}
                    resizeMode="contain"
                    source={
                      Object {
                        "uri": "https://images.ctfassets.net/6jsfqc13oxv4/28yQznd8Yw0jugpXqSteow/65a161d411d1920a303a5d5c7a65e424/banner-card_app-polos.jpg",
                      }
                    }
                    style={
                      Array [
                        Object {
                          "height": 0,
                          "width": 621.5,
                        },
                        undefined,
                      ]
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
      <View
        alignSelf="center"
        flexDirection="row"
        height={24}
        style={
          Array [
            Object {
              "alignSelf": "center",
              "flexDirection": "row",
              "height": 24,
            },
          ]
        }
      >
        <View
          accessibilityLabel="com.usereserva:id/cards_carrousel_animated_view"
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#6F6F6F",
              "borderRadius": 7,
              "height": 7,
              "position": "absolute",
              "transform": Array [
                Object {
                  "translateX": 6,
                },
              ],
              "width": 7,
              "zIndex": 2,
            }
          }
          testID="com.usereserva:id/cards_carrousel_animated_view"
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "width": 19,
            }
          }
        >
          <View
            style={
              Object {
                "borderColor": "#6F6F6F",
                "borderRadius": 7,
                "borderWidth": 1,
                "height": 7,
                "width": 7,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "width": 19,
            }
          }
        >
          <View
            style={
              Object {
                "borderColor": "#6F6F6F",
                "borderRadius": 7,
                "borderWidth": 1,
                "height": 7,
                "width": 7,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "width": 19,
            }
          }
        >
          <View
            style={
              Object {
                "borderColor": "#6F6F6F",
                "borderRadius": 7,
                "borderWidth": 1,
                "height": 7,
                "width": 7,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
